### PR TITLE
[ENG-434] No throttle user email gets

### PR DESCRIPTION
**Pull request by jamescdavis**
## Purpose

The Emberized account settings page needs to be able to get the user emails list without being throttled. The `SendEmailThrottle` throttling class was applied to the `UserEmailsList` view to throttle requests that result in an email being sent. `GET` requests to this endpoint, however, do not send an email and shouldn't be throttled. This PR bypassing throttling for `GET` requests to endpoints using the `SendEmailThrottle` throttling class. These include the `UserEmailsList` view as well as `UserAccountExport`, for which `GET` is not implemented. In addition, the `SendEmailDeactivationThrottle` throttling class inherits from `SendEmailThrottle`, which only sends an email when `PATCH`ing `deactivation_requested`.

## Changes

* override `allow_request` method on `SendEmailThrottle` class to always allow `GET` requests to pass through unthrottled.
* add a test to make sure that three `GET` requests to user emails list endpoint in rapid succession do not result in throttling (the default throttling for `send-email` is `2/minute`, so the third request would be throttled without the `GET` exception)

## QA Notes

Make sure multiple requests in rapid succession to `/v2/users/me/settings/emails/` are not throttled.

## Documentation

n/a because this does not change API surface

## Side Effects

Should be none.

## Ticket

https://openscience.atlassian.net/browse/ENG-434

### CODE REVIEW

**Comment by Johnetordoff** [comment]
Did Yoda write the title of this PR?

**Comment by jamescdavis** [comment]
```python
if true == write_yoda_did:
    condition = 'Yoda'
```

File: api/base/throttling.py [hunk 1]
```diff
@@ -101,6 +101,11 @@ class SendEmailThrottle(BaseThrottle, UserRateThrottle):
 
     scope = 'send-email'
 
+    def allow_request(self, request, view):
+        if request.method == 'GET':
```
Comment (pattisdr) [inline api/base/throttling.py]: I'm wondering if we should broaden this so 'OPTIONS' request method is included - 

```python
from rest_framework import permissions

def allow_request(self, request, view):
      if request.method in permissions.SAFE_METHODS:
            return True
```

Comment (jamescdavis) [reply api/base/throttling.py]: Oh yes, totally agree! And Ember does an `OPTIONS` request first so this could definitely be an issue.

Comment (jamescdavis) [reply api/base/throttling.py]: Updated!

**Pull request merged**